pkg/manifests/manifests.go

```diff
@@ -79,17 +79,15 @@ func (f *Factory) RouterDaemonSet(cr *ingressv1alpha1.ClusterIngress) (*appsv1.D
 
 	ds.Name = name
 
-	ds.Spec.Template.Labels = map[string]string{
-		"app":    "router",
-		"router": name,
+	if ds.Spec.Template.Labels == nil {
+		ds.Spec.Template.Labels = map[string]string{}
 	}
+	ds.Spec.Template.Labels["router"] = name
 
-	ds.Spec.Selector = &metav1.LabelSelector{
-		MatchLabels: map[string]string{
-			"app":    "router",
-			"router": name,
-		},
+	if ds.Spec.Selector.MatchLabels == nil {
+		ds.Spec.Selector.MatchLabels = map[string]string{}
 	}
+	ds.Spec.Selector.MatchLabels["router"] = name
```

Comment:
We should update `assets/router/service-cloud.yaml` to reflect that only the one field, and not the entirety of `.spec.selector`, is replaced at runtime.

Reply:
Whoops, I commented in the wrong place.